src/lib/Microsoft.Health.Events/Repository/StorageManager.cs

```diff
@@ -0,0 +1,38 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.IO;
+using Azure.Storage.Blobs;
+using EnsureThat;
+
+namespace Microsoft.Health.Events.Repository
+{
+    public class StorageManager : IRepositoryManager
+    {
+        private BlobContainerClient _blobContainer;
+
+        public StorageManager(string connectionString, string blobContainerName)
+        {
+            EnsureArg.IsNotNull(connectionString);
+            EnsureArg.IsNotNull(blobContainerName);
+
+            _blobContainer = new BlobContainerClient(connectionString, blobContainerName);
```

Comment:
We use `BlobContainerClient` in several places. Can we pass in a shared instance via the constructor versus creating new ones? Would also help with testing as we could simply pass in a mocked instance. And your validation logic for the params could be centralized in one place.

Reply:
It is a good idea. One issue is that I believe BlobContainerClient is specific to just one blob BlobContainer. Currently we have one blob container for templates called "template" and I didn't want to put my checkpoints in there. So for checkpoints I created another container and concluded that I needed another BlobContainerClient. I was really wanting to consolidate but I think there may be limitations. I will take another looks to see what can be done in case I missed something.